services/sonar/sonar-coverage.tester.js

```diff
@@ -0,0 +1,22 @@
+'use strict'
+
+const t = (module.exports = require('../tester').createServiceTester())
+const { isIntegerPercentage } = require('../test-validators')
+
+t.create('Coverage')
+  .get(
+    '/http/sonar.petalslink.com/org.ow2.petals%3Apetals-se-ase/coverage.json'
+  )
+  .expectBadge({
+    label: 'coverage',
+    message: isIntegerPercentage,
+  })
+
+t.create('Coverage (legacy API supported)')
+  .get(
+    '/http/sonar.petalslink.com/org.ow2.petals%3Apetals-se-ase/coverage.json?version=4.2'
+  )
+  .expectBadge({
+    label: 'coverage',
+    message: isIntegerPercentage,
+  })
```

Comment:
Quite a few of these test files don't include a test for the "not found" case. Given we're supporting lots of versions of an API that may be self- hosted, I think the role of the 'not found' test as a canary for an upstream API changes is less clear, but do you reckon its worth adding one test just on the base class that hits `sonarcloud.io`?

Reply:
My personal preference would have been to include a `not found` case in each service tester, but in some previous PR reviews there was pushback on those cases from other maintainers when the services all used the same `fetch` functions in the same base class. 

As such I only added one live test case validating the `not found` scenario on sonarcloud.io that is in the `fortify` [tester](https://github.com/badges/shields/blob/256c0d07664e332bf7fb360803f4de2a2b45af1d/services/sonar/sonar-fortify-rating.tester.js#L83) file